oximeter/collector/src/self_stats.rs

```diff
@@ -9,51 +9,21 @@
 use crate::ProducerEndpoint;
 use oximeter::types::Cumulative;
 use oximeter::types::ProducerResultsItem;
-use oximeter::Metric;
 use oximeter::MetricsError;
 use oximeter::Sample;
-use oximeter::Target;
 use reqwest::StatusCode;
 use std::borrow::Cow;
 use std::collections::BTreeMap;
-use std::net::IpAddr;
 use std::time::Duration;
-use uuid::Uuid;
+
+oximeter::use_timeseries!("oximeter-collector.toml");
+pub use self::oximeter_collector::Collections;
+pub use self::oximeter_collector::FailedCollections;
+pub use self::oximeter_collector::OximeterCollector;
 
 /// The interval on which we report self statistics
 pub const COLLECTION_INTERVAL: Duration = Duration::from_secs(60);
 
-/// A target representing a single oximeter collector.
-#[derive(Clone, Copy, Debug, Target)]
-pub struct OximeterCollector {
-    /// The collector's ID.
-    pub collector_id: Uuid,
-    /// The collector server's IP address.
-    pub collector_ip: IpAddr,
-    /// The collector server's port.
-    pub collector_port: u16,
-}
-
-/// The number of successful collections from a single producer.
-#[derive(Clone, Debug, Metric)]
-pub struct Collections {
-    /// The producer's ID.
-    pub producer_id: Uuid,
-    /// The producer's IP address.
-    pub producer_ip: IpAddr,
-    /// The producer's port.
-    pub producer_port: u16,
-    /// The base route in the producer server used to collect metrics.
-    ///
-    /// The full route is `{base_route}/{producer_id}`.
-    ///
-    // TODO-cleanup: This is no longer relevant, but removing it entirely
-    // relies on nonexistent functionality for updating timeseries schema. When
```

Comment:
Should this TODO be copied over? Same for `FailedCollections`

Reply:
Good question, I made https://github.com/oxidecomputer/omicron/issues/6028 to keep track of that. Thanks!